### DISCUSSION TITLE
How to get per-example gradients in Flax?

### DISCUSSION THREAD

**Discussion by marcvanzee** [google/flax#2084] [Q&A] [answered]
How to get per-example gradients in Flax?

**Comment by marcvanzee** [comment, marked as answer]
All layers accept batch dimension free inputs. Example:

```py
def loss_fn(param, batch):
  return jnp.square(param * batch)

model = 3.
per_example_grad_fn = jax.vmap(jax.grad(loss_fn), in_axes=(None, 0))
per_example_grad_fn(param, jnp.ones((3,)))
```

**Reply by mshnmshn** [reply]
It would be nice to have some `linen` layers in the example like in this [PR](https://github.com/google/flax/pull/306/commits/01344357224901fed3e7c474de5398c410639f11).

**Reply by VMS-6511** [reply]
Are there any examples of this with `linen` layers as mentioned above?
